server/routes/request-book.ts

```diff
@@ -0,0 +1,67 @@
+import { Router } from "express";
+import renderTemplate from "../utils/renderTemplate";
+import { requireAuth } from "../middlewares/auth";
+import { User } from "../entities/User";
+import BookRequest from "../entities/BookRequest";
+
+type BookRequestBody = {
+  bookTitle: string;
+  bookAuthor: string;
+  noteRequest?: string;
+};
+
+let router = Router();
+
+router.get("/", requireAuth, function (req, res) {
+  let currentUser = res.locals.user as User;
+  if (!currentUser.isTeacher) {
+    res.redirect("/");
+    return;
+  }
+  renderTemplate(req, res, "request-book.html", {
+    title: "Yêu cầu thêm sách",
+  });
+});
+
+router.post("/", requireAuth, async function (req, res) {
+  let currentUser = res.locals.user as User;
+  if (!currentUser.isTeacher) {
+    res.redirect("/");
+    return;
+  }
+
+  let { bookTitle, bookAuthor, noteRequest }: BookRequestBody = req.body;
+
+  try {
+    if (
+      typeof bookTitle !== "string" ||
+      typeof bookAuthor !== "string" ||
+      typeof noteRequest !== "string"
+    ) {
+      throw new Error("Don't do that 1");
```

Comment:
bỏ cái 1 đi?

Reply:
Khi hồi code nó trả về Don't do that mà đéo biết có phải chắc chắn cái này không thêm vào để nhận diện :)